flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -304,7 +306,15 @@ enum DataType {
         RECOVERY_COMPLETION(false, true, true, false, false),
 
         /** {@link #END_OF_SEGMENT} indicates that a segment is finished in a subpartition. */
-        END_OF_SEGMENT(false, true, false, false, false);
+        END_OF_SEGMENT(false, true, false, false, false),
+
+        /**
+         * {@link #END_OF_DATA} indicates that there will be no more data buffer in a subpartition.
+         */
+        END_OF_DATA(false, true, false, false, false),
```

Comment:
We already had `EndOfData` and `EndOfPartitionEvent` events, and why do we can not reuse them here?

Reply:
EndOfData and EndOfPartitionEvent are RuntimeEvent subclasses, while here we need Buffer.DataType enum values.